### ISSUE TITLE
Which ImageNet-100?

### ISSUE THREAD

**Issue by YugeTen** [vturrisi/solo-learn#137]
Hi all,

First of all, thank you so much for creating this library, I have found it to be super useful for my own research!

I was wondering if you could provide some details on the ImageNet-100 dataset that you used? I cannot seem to find any "standard" ImageNet-100 dataset for downloading on the internet and the papers that use this dataset (eg. [1](https://arxiv.org/pdf/2104.03952v1.pdf) and [2](https://arxiv.org/pdf/2005.12320.pdf)) seem to randomly select 100 classes from the dataset.

Any of your help would be much appreciated!

**Comment by vturrisi** [comment]
Hey,

Thanks for the comments! We are glad that this is useful for other researchers :)
About imagenet-100, you can get the list of classes from [here](https://github.com/HobbitLong/CMC/issues/21).
We will probably add this list to our repo somewhere to avoid confusion.

**Comment by ankitpatnala** [comment]
> Hi all,
> 
> First of all, thank you so much for creating this library, I have found it to be super useful for my own research!
> 
> I was wondering if you could provide some details on the ImageNet-100 dataset that you used? I cannot seem to find any "standard" ImageNet-100 dataset for downloading on the internet and the papers that use this dataset (eg. [1](https://arxiv.org/pdf/2104.03952v1.pdf) and [2](https://arxiv.org/pdf/2005.12320.pdf)) seem to randomly select 100 classes from the dataset.
> 
> Any of your help would be much appreciated!

This repo is very useful [tiny image-net](https://github.com/mf1024/ImageNet-Datasets-Downloader) to download images

**Comment by vturrisi** [comment]
thanks, I'll check it out

**Comment by YugeTen** [comment]
Hi @vturrisi, thanks for the swift response! Apologies, I should have done a more thorough search of the closed issues!

One more question: do you use all the images in these 100 classes?

@ankitpatnala thanks for sharing the repo :)

**Comment by vturrisi** [comment]
@YugeTen no worries! And yes, we use all the images, both for the self-supervised pertaining and the linear classifier.

**Comment by YugeTen** [comment]
Thank you!

**Comment by YingYingFight** [comment]
> Hey,
> 
> Thanks for the comments! We are glad that this is useful for other researchers :) About imagenet-100, you can get the list of classes from [here](https://github.com/HobbitLong/CMC/issues/21). We will probably add this list to our repo somewhere to avoid confusion.

@vturrisi Hi! Thank you for your update of pretrained checkpoints in the model zoo recently! I hope to use the checkpoint on ImageNet-100 for some evaluations, and I have downloaded the dataset as you said from [here](https://github.com/HobbitLong/CMC/issues/21). However, I still have a question:

What's the order of class in the final 100-dim predictions? I mean, with  the pretained checkpoint in the model zoo,  what class is  each dimension in the final 100-dim prediction logit?

Thank you very much!

**Comment by vturrisi** [comment]
Hey @YingYingFight 
Just sort the labels and you will get the correct order, e.g., `labels = sorted(Path(entry.name) for entry in os.scandir(data_path) if entry.is_dir())`.

**Comment by YingYingFight** [comment]
> name
@vturrisi Thank you for your reply!

I wonder does the `entry.name` is like `n02869837` or the class name like `dog`?

Thank you!

**Comment by vturrisi** [comment]
it's the former `(n02869837)`

**Comment by YingYingFight** [comment]
> it's the former `(n02869837)`.

@vturrisi 

Got it! Thank you very much for your help!

**Comment by vturrisi** [comment]
No worries! Thanks for using the library :)
